### DISCUSSION TITLE
Any way to pass a data object to element/component slots?

### DISCUSSION THREAD

**Discussion by pierrecdn** [zauberzeug/nicegui#2440] [Q&A]
### Question

Hi :wave:  and thanks for this awesome framework and abstraction!

Do we have a way to define data functions when adding slots? I'm especially wondering how to initialize and interact with a `q-checkbox` element initialized within a `ui.table.add_slot` context, but it can probably be generalized to a lot of other use-cases.

Example:
```py
table = ui.table(**stuff)
table.add_slot(
    "top",
    f"""
    <q-checkbox :checked-icon="'star'" :unchecked-icon="'star_border'" color='grey'>
        <q-tooltip>Filter elements based on {criteria}</q-tooltip>
    </q-checkbox>
    """,
)
```

* First, I'm getting an uninitialized checkbox (indetermined state). How can I deal with that without the ability to pass the [recommended](https://v2.vuejs.org/v2/style-guide/?redirect=true#Component-data-essential) `data` function?
* Then, I'll need to retrieve values on update. According to [quasar docs](https://quasar.dev/vue-components/checkbox), I'd say that adding that `@update:model-value` property and register the event handler would do the job?

```
   <q-checkbox :checked-icon="'star'" :unchecked-icon="'star_border'" color='grey' @update:model-value="$parent.$emit('filter_event')">
(...)
table.on('filter_event", handler=filter_callback)
```

Nota: I ended up reading [this](https://github.com/zauberzeug/nicegui/discussions/1138#discussioncomment-6383197), but maybe you'd have some other suggestion to provide in my (simple?) case.

**Comment by falkoschindler** [comment]
Hi @pierrecdn,

You can initialize the checkbox using the `v-model` or `:model-value` attribute.
I just posted a similar example over here: https://github.com/zauberzeug/nicegui/discussions/2437#discussioncomment-8253673.

But I still wish there was an easier way to interact with slot templates.

**Reply by pierrecdn** [reply]
In the given example, it was a "cell" slot, which imply there's a mapping (through the rows) to the corresponding cell value.

I don't get how I can transpose this example to my "top" slot. The goal in the end is to apply `.filter` on the table when clicked.

**Reply by falkoschindler** [reply]
Ah ok. It should work something like this:
```py
columns = [{'name': 'name', 'label': 'Name', 'field': 'name'}]
rows = [{'name': 'Alice'}, {'name': 'Bob'}, {'name': 'Charlie'}]

def handle_filter(e: events.GenericEventArguments) -> None:
    ui.notify(f'Filter: {e.args}')

table = ui.table(columns=columns, rows=rows, row_key='name')
table.add_slot('top', r'''
    <div :props="props">
        <q-checkbox :model-value="true" @update:model-value="(value) => $parent.$emit('filter', value)" />
    </div>
''')
table.on('filter', handle_filter)
```

Unfortunately the checkbox doesn't update in the UI, even though the even is emitted. `table.update()` doesn't help, because it doesn't affect the "top" slot. Does anyone have an idea how to solve this?
